clang/lib/Driver/Driver.cpp

```diff
@@ -7984,8 +7998,15 @@ void Driver::BuildJobs(Compilation &C) const {
     // DiagnosticsEngine, so that extra values, position, and so on could be
     // printed.
     if (!A->isClaimed()) {
-      if (A->getOption().hasFlag(options::NoArgumentUnused))
-        continue;
+      if (A->getOption().hasFlag(options::NoArgumentUnused)) {
+        if ((A->getOption().matches(
+                 options::OPT_fsycl_dead_args_optimization) ||
+             A->getOption().matches(options::OPT_fsycl_device_lib_jit_link)) &&
+            !HasValidSYCLRuntime)
+          ;
+        else
+          continue;
+      }
```

Comment:
Instead of this - it's easier just to remove the `NoArgumentUnused` from the `Options.td` file